File: momentra/momentra/Shell/GroupActive/Shared/GroupPulseInsightsCard.swift
import SwiftUI

/// Live group pulse insights from `GET /v1/analytics/insights` (scopeType=MOMENT).
struct GroupPulseInsightsHeroCard: View {
    let headerTitle: String
    let insights: [AnalyticsInsightItemPayload]
    let gradient: LinearGradient
    var titleColor: Color = .white
    var bodyColor: Color = Color.white.opacity(0.9)
    var footerLabel: String?
    var footerAction: (() -> Void)?

    var body: some View {
        VStack(alignment: .leading, spacing: 12) {
            Text(headerTitle)
                .font(.plusJakarta(size: 18, weight: .heavy))
                .foregroundStyle(titleColor)

            if insights.isEmpty {
                Text("No insights yet")
                    .font(.plusJakarta(size: 13, weight: .semibold))
                    .foregroundStyle(titleColor)
                Text("Insights appear when there's enough group activity — nothing is invented.")
                    .font(.plusJakarta(size: 12))
                    .foregroundStyle(bodyColor)
            } else {
                ForEach(Array(insights.prefix(3)), id: \.insightId) { item in
                    VStack(alignment: .leading, spacing: 4) {
                        if let title = item.title, !title.isEmpty {
                            Text(title)
                                .font(.plusJakarta(size: 13, weight: .semibold))
                                .foregroundStyle(titleColor)
                        }
                        if let body = item.body, !body.isEmpty {
                            Text(body)
                                .font(.plusJakarta(size: 12))
                                .foregroundStyle(bodyColor)
                        }
                    }
                }
            }

            if let footerLabel, let footerAction {
                Button(action: footerAction) {
                    Text(footerLabel)
                        .font(.plusJakarta(size: 14, weight: .bold))
                        .foregroundStyle(Color(hex: "#131313"))
                        .frame(maxWidth: .infinity)
                        .padding(.vertical, 14)
                        .background(Color.white)
                        .clipShape(RoundedRectangle(cornerRadius: 12))
                }
                .buttonStyle(.plain)
            }
        }
        .padding(20)
        .frame(maxWidth: .infinity, alignment: .leading)
        .background(gradient)
        .clipShape(RoundedRectangle(cornerRadius: 18))
    }
}

struct GroupPulseInsightsSectionCard: View {
    let insights: [AnalyticsInsightItemPayload]

    var body: some View {
        GroupSectionCard(title: "Momentra Insights") {
            if insights.isEmpty {
                GroupEmptySection(
                    message: "No insights yet",
                    detail: "Insights appear when there's enough group activity — nothing is invented."
                )
            } else {
                VStack(alignment: .leading, spacing: 12) {
                    ForEach(Array(insights.prefix(3)), id: \.insightId) { item in
                        VStack(alignment: .leading, spacing: 4) {
                            if let title = item.title, !title.isEmpty {
                                Text(title)
                                    .font(.plusJakarta(size: 13, weight: .semibold))
                                    .foregroundStyle(GroupActiveTheme.text)
                            }
                            if let body = item.body, !body.isEmpty {
                                Text(body)
                                    .font(.plusJakarta(size: 12))
                                    .foregroundStyle(GroupActiveTheme.secondary)
                            }
                        }
                    }
                }
            }
        }
    }
}

File: apk/app/src/main/java/com/example/momentra/ui/shell/group/shared/GroupPulseInsightsCard.kt
package com.example.momentra.ui.shell.group.shared

import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.example.momentra.data.api.AnalyticsInsightItemDto
import com.example.momentra.ui.theme.PlusJakartaSans

@Composable
fun GroupPulseInsightsHeroCard(
    headerTitle: String,
    insights: List<AnalyticsInsightItemDto>,
    gradient: Brush,
    titleColor: Color = Color.White,
    bodyColor: Color = Color.White.copy(alpha = 0.9f),
    footerLabel: String? = null,
    onFooterClick: (() -> Unit)? = null,
    modifier: Modifier = Modifier,
) {
    Column(
        modifier = modifier
            .fillMaxWidth()
            .clip(RoundedCornerShape(18.dp))
            .background(gradient)
            .padding(20.dp),
        verticalArrangement = Arrangement.spacedBy(12.dp),
    ) {
        Text(
            headerTitle,
            color = titleColor,
            fontSize = 18.sp,
            fontWeight = FontWeight.ExtraBold,
            fontFamily = PlusJakartaSans,
        )
        if (insights.isEmpty()) {
            Text(
                "No insights yet",
                color = titleColor,
                fontSize = 13.sp,
                fontWeight = FontWeight.SemiBold,
                fontFamily = PlusJakartaSans,
            )
            Text(
                "Insights appear when there's enough group activity — nothing is invented.",
                color = bodyColor,
                fontSize = 12.sp,
                fontFamily = PlusJakartaSans,
            )
        } else {
            insights.take(3).forEach { item ->
                Column(verticalArrangement = Arrangement.spacedBy(4.dp)) {
                    item.title?.takeIf { it.isNotBlank() }?.let {
                        Text(
                            it,
                            color = titleColor,
                            fontSize = 13.sp,
                            fontWeight = FontWeight.SemiBold,
                            fontFamily = PlusJakartaSans,
                        )
                    }
                    item.body?.takeIf { it.isNotBlank() }?.let {
                        Text(
                            it,
                            color = bodyColor,
                            fontSize = 12.sp,
                            fontFamily = PlusJakartaSans,
                        )
                    }
                }
            }
        }
        if (footerLabel != null && onFooterClick != null) {
            Text(
                footerLabel,
                color = Color(0xFF131313),
                fontSize = 14.sp,
                fontWeight = FontWeight.Bold,
                fontFamily = PlusJakartaSans,
                modifier = Modifier
                    .fillMaxWidth()
                    .clip(RoundedCornerShape(12.dp))
                    .background(Color.White)
                    .clickable(onClick = onFooterClick)
                    .padding(vertical = 14.dp),
            )
        }
    }
}

@Composable
fun GroupPulseInsightsSectionCard(
    insights: List<AnalyticsInsightItemDto>,
    modifier: Modifier = Modifier,
) {
    GroupSectionCard(title = "Momentra Insights", modifier = modifier) {
        if (insights.isEmpty()) {
            GroupEmptySection(
                message = "No insights yet",
                detail = "Insights appear when there's enough group activity — nothing is invented.",
            )
        } else {
            Column(verticalArrangement = Arrangement.spacedBy(12.dp)) {
                insights.take(3).forEach { item ->
                    Column(verticalArrangement = Arrangement.spacedBy(4.dp)) {
                        item.title?.takeIf { it.isNotBlank() }?.let {
                            Text(
                                it,
                                color = GroupActiveTheme.Text,
                                fontSize = 13.sp,
                                fontWeight = FontWeight.SemiBold,
                                fontFamily = PlusJakartaSans,
                            )
                        }
                        item.body?.takeIf { it.isNotBlank() }?.let {
                            Text(
                                it,
                                color = GroupActiveTheme.Secondary,
                                fontSize = 12.sp,
                                fontFamily = PlusJakartaSans,
                            )
                        }
                    }
                }
            }
        }
    }
}
